### ISSUE TITLE
Linux and Darwin builds fail to execute (panic at runtime)

### ISSUE THREAD

**Issue by rvolosatovs** [burrito-elixir/burrito#89]
### Host

Tell us about your Host OS (the machine you're building burrito apps on)

**Host OS**: NixOS
**Host CPU**: x86_64

Please run `zig version` in your Burrito project directory and write the version here:
0.10.1

Please run `elixir -v` in your Burrito project directory and write the version here:
```
Erlang/OTP 25 [erts-13.2] [source] [64-bit] [smp:16:8] [ds:16:8:10] [async-threads:1] [jit:ns]

Elixir 1.14.4 (compiled with Erlang/OTP 25)
```
------

### Target

**Host OS**: Alpine (`alpine:latest` docker image)
**Host CPU**: x86_64
**Host LibC and Version**
```
$ ldd --version
musl libc (x86_64)
Version 1.2.3
Dynamic Program Loader
Usage: /lib/ld-musl-x86_64.so.1 [options] [--] pathname
```

------

```
# lots of unpacked files...
debug: Unpacked File: /root/.local/share/.burrito/host_core_erts-13.2_0.62.1/lib/runtime_tools-1.19/priv/lib/dyntrace.so
debug: Unpacked File: /root/.local/share/.burrito/host_core_erts-13.2_0.62.1/lib/runtime_tools-1.19/priv/lib/trace_file_drv.so
debug: Unpacked File: /root/.local/share/.burrito/host_core_erts-13.2_0.62.1/lib/runtime_tools-1.19/priv/lib/trace_ip_drv.so
debug: Unpacked File: /root/.local/share/.burrito/host_core_erts-13.2_0.62.1/lib/.burrito
thread 61 panic: attempt to use null value
/home/rvolosatovs/src/github.com/wasmCloud/wasmcloud-otp/host_core/deps/burrito/src/archiver.zig:198:49: 0x21d09c2 in unpack_files (host_core)
        try create_dirs(dest_path[0..], dir_name.?, allocator);
                                                ^
/home/rvolosatovs/src/github.com/wasmCloud/wasmcloud-otp/host_core/deps/burrito/src/wrapper.zig:148:69: 0x21d1304 in do_payload_install (host_core)
    try foilz.unpack_files(FOILZ_PAYLOAD, install_dir, build_options.UNCOMPRESSED_SIZE);
                                                                    ^
/home/rvolosatovs/src/github.com/wasmCloud/wasmcloud-otp/host_core/deps/burrito/src/wrapper.zig:114:31: 0x21db209 in main (host_core)
        try do_payload_install(install_dir, metadata_path);
                              ^
/nix/store/50dflzl0472gwzpnwhxylnj4a4a0i6q7-zig-0.10.1/lib/zig/std/start.zig:614:37: 0x21dc05b in main (host_core)
???:?:?: 0x2270e11 in ??? (???)
Aborted (core dumped)
```

Note that I just tried the Gnu build in Ubuntu container with the exact same result:
```
thread 9 panic: attempt to use null value
/home/rvolosatovs/src/github.com/wasmCloud/wasmcloud-otp/host_core/deps/burrito/src/archiver.zig:198:49: 0x17818e9 in unpack_files (host_core)
        try create_dirs(dest_path[0..], dir_name.?, allocator);
                                                ^
/home/rvolosatovs/src/github.com/wasmCloud/wasmcloud-otp/host_core/deps/burrito/src/wrapper.zig:148:69: 0x1782221 in do_payload_install (host_core)
    try foilz.unpack_files(FOILZ_PAYLOAD, install_dir, build_options.UNCOMPRESSED_SIZE);
                                                                    ^
/home/rvolosatovs/src/github.com/wasmCloud/wasmcloud-otp/host_core/deps/burrito/src/wrapper.zig:114:31: 0x178bfdb in main (host_core)
        try do_payload_install(install_dir, metadata_path);
                              ^
/nix/store/50dflzl0472gwzpnwhxylnj4a4a0i6q7-zig-0.10.1/lib/zig/std/start.zig:614:37: 0x178cde7 in main (host_core)
Aborted (core dumped)
```

arm64 Darwin:
```
thread 7197710 panic: attempt to use null value
???:?:?: 0x10440da07 in _archiver.unpack_files (???)
???:?:?: 0x10440e473 in _wrapper.do_payload_install (???)
???:?:?: 0x104417cab in _wrapper.main (???)
???:?:?: 0x104418a07 in _main (???)
???:?:?: 0x197ebfe4f in ??? (???)
???:?:?: 0x2b5fffffffffffff in ??? (???)
[1]    13103 abort      ./host_core_aarch64_darwin
```

**Comment by doawoo** [comment]
I'm going to dive deeper into trying to reproduce these later, but for a quick check, we noticed in contained that, due to the lack of a proper `HOME` burrito can't deduce where to place files when they're extracted.

Does setting the environment variable `HOST_CORE_INSTALL_DIR` to a known, writeable path, help when trying to run?

Note the `XXX_INSTALL_DIR` is computed based on the release name, I noticed you release was called `host_core` so I assumed that would be the proper env variable to set.

Zig code: https://github.com/burrito-elixir/burrito/blob/main/src/wrapper.zig#LL159C69-L159C84
